applications/CompressiblePotentialFlowApplication/python_scripts/apply_far_field_process.py

```diff
@@ -24,20 +29,32 @@ def __init__(self, Model, settings ):
         settings.ValidateAndAssignDefaults(default_parameters);
 
         self.model_part = Model[settings["model_part_name"].GetString()]
-        self.velocity_infinity = KratosMultiphysics.Vector(3)#array('d', [1.0, 2.0, 3.14])#np.array([0,0,0])#np.zeros(3)#vector(3)
-        self.velocity_infinity[0] = settings["velocity_infinity"][0].GetDouble()
-        self.velocity_infinity[1] = settings["velocity_infinity"][1].GetDouble()
-        self.velocity_infinity[2] = settings["velocity_infinity"][2].GetDouble()
-        #self.density_infinity = settings["density_infinity"].GetDouble() #TODO: must read this from the properties
+        self.fluid_model_part = self.model_part.GetRootModelPart()
+
+        self.angle_of_attack = settings["angle_of_attack"].GetDouble()
+        self.mach_inf = settings["mach_infinity"].GetDouble()
+        self.density_inf = settings["density_infinity"].GetDouble()
+        self.seed_of_sound_inf = settings["speed_of_sound"].GetDouble()
+        self.heat_capacity_ratio = settings["heat_capacity_ratio"].GetDouble()
         self.inlet_phi = settings["inlet_phi"].GetDouble()
-        self.model_part.ProcessInfo.SetValue(CompressiblePotentialFlowApplication.VELOCITY_INFINITY,self.velocity_infinity)
 
+        # Computing free stream velocity
+        self.u_inf = self.mach_inf * self.seed_of_sound_inf
+        self.velocity_infinity = KratosMultiphysics.Vector(3)
+        self.velocity_infinity[0] = round(self.u_inf*math.cos(self.angle_of_attack),8)
+        self.velocity_infinity[1] = round(self.u_inf*math.sin(self.angle_of_attack),8)
+        self.velocity_infinity[2] = 0.0
 
+        self.fluid_model_part.ProcessInfo.SetValue(CPFApp.MACH_INFINITY,self.mach_inf)
+        self.fluid_model_part.ProcessInfo.SetValue(CPFApp.VELOCITY_INFINITY,self.velocity_infinity)
+        self.fluid_model_part.ProcessInfo.SetValue(CPFApp.DENSITY_INFINITY,self.density_inf)
+        self.fluid_model_part.ProcessInfo.SetValue(KratosMultiphysics.SOUND_VELOCITY,self.seed_of_sound_inf)
+        self.fluid_model_part.ProcessInfo.SetValue(CPFApp.HEAT_CAPACITY_RATIO,self.heat_capacity_ratio)
 
     def Execute(self):
-        #KratosMultiphysics.VariableUtils().SetVectorVar(CompressiblePotentialFlowApplication.VELOCITY_INFINITY, self.velocity_infinity, self.model_part.Conditions)
+        #KratosMultiphysics.VariableUtils().SetVectorVar(CPFApp.VELOCITY_INFINITY, self.velocity_infinity, self.model_part.Conditions)
         for cond in self.model_part.Conditions:
-            cond.SetValue(CompressiblePotentialFlowApplication.VELOCITY_INFINITY, self.velocity_infinity)
+            cond.SetValue(CPFApp.VELOCITY_INFINITY, self.velocity_infinity)
```

Comment:
why is it needed in every condition?

Reply:
Here we loop only over the conditions of the far field. In all of the conditions of the far field we need to apply the boundary term so we set the variable VELOCITY_INFINITY to the velocity in the free stream which is then retrieved in the condition:

https://github.com/KratosMultiphysics/Kratos/blob/3991986010f04c22a8de9179feaac52233f4c39d/applications/CompressiblePotentialFlowApplication/custom_conditions/potential_wall_condition.cpp#L105-L107

Since we do not set this variable for the body, the body has zero contribution to the Neumann term (as desired). Is there a better way of doing this?

Reply:
One option would be to define v_inf in the ProcessInfo, but we would need to use flags to differentiate between body and far_field in the condition .cpp. Is this preferred over what we have right now?

Reply:
This is a good idea. However, I would rather avoid including more if statements in the conditions and the elements. I believe we are already using them too much. But lets see what the others say.